pynicotine/slskproto.py

```diff
@@ -217,22 +216,34 @@
 UINT_PACK = struct.Struct("<I").pack
 
 
+class ConnectionInProgress:
+    """ As all p2p connect()s are non-blocking, this class is used to
+    hold data about a connection that is not yet established """
+
+    __slots__ = ("sock", "addr", "lastactive", "init", "login")
+
+    def __init__(self, sock=None, addr=None, init=None, login=None):
+        self.sock = sock
+        self.addr = addr
+        self.init = init
+        self.login = login
+        self.lastactive = time.time()
+
+
 class Connection:
-    """
-    Holds data about a connection. conn is a socket object,
+    """ Holds data about a connection. sock is a socket object,
     addr is (ip, port) pair, ibuf and obuf are input and output msgBuffer,
-    init is a PeerInit object (see slskmessages docstrings).
-    """
+    init is a PeerInit object (see slskmessages docstrings). """
 
-    __slots__ = ("conn", "addr", "ibuf", "obuf", "events", "init", "lastactive", "lastreadlength")
+    __slots__ = ("sock", "addr", "init", "ibuf", "obuf", "events", "lastactive", "lastreadlength")
 
-    def __init__(self, conn=None, addr=None, events=None):
-        self.conn = conn
+    def __init__(self, sock=None, addr=None, events=None):
+        self.sock = sock
```

Comment:
Consider `link` instead of `sock` , because sock is ambiguous with filesystem data 'sockets'.

Reply:
I think `link` is even more ambiguous (at least I've never heard the term in this context).

Reply:
How about `net`

Reply:
I'll keep the current name, at least for now. Already spent too many hours renaming everything, and I'd rather not go through that again.